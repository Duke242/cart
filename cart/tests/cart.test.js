
import React from "react"
import { act, render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Home from "../pages/index"
import { getByTagName } from "@testing-library/dom";
import App from '../pages/_app'



describe("App component", () => {
  it("h1 reads SHOP", () => {
    render(<Home />)

    expect(screen.getByRole("heading").textContent).toMatch(/SHOP/)
  })
})

describe('Cart Componenet', () => { 
  it('increments cart total', () => {
    const { container } = render(<App Component={Home}/>)
    const button = container.querySelector('.addButton');
    act(() => {
      button.click()
    })
    console.log(container.textContent)
    expect(container.querySelector('h1').textContent).toMatch(/SHOP: 1/)
  })
})