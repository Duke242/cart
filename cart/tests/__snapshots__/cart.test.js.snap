// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div>
  <div
    class="description"
  >
    <h1>
      SHOP
    </h1>
  </div>
  <ul>
    <li>
      Monitor
       ($
      1000.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Keyboard
       ($
      50.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Mouse
       ($
      20.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Storage
       ($
      60.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
  </ul>
</div>
`;

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <div
    class="description"
  >
    <h1>
      SHOP
    </h1>
  </div>
  <ul>
    <li>
      Monitor
       ($
      1000.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Keyboard
       ($
      50.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Mouse
       ($
      20.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
    <li>
      Storage
       ($
      60.00
      )
      <button>
        ðŸ›’
      </button>
    </li>
  </ul>
</div>
`;
